### PR TITLE
Bugfix/race conditions

### PR DESCRIPTION
**Pull request by sp-marcos-quesada**

### CODE REVIEW

File: text_formatter.go [hunk 1]
```diff
@@ -73,10 +82,12 @@ func (f *TextFormatter) Format(entry *Entry) ([]byte, error) {
 	isColorTerminal := isTerminal && (runtime.GOOS != "windows")
 	isColored := (f.ForceColors || isColorTerminal) && !f.DisableColors
 
+	f.Mutex.Lock()
 	timestampFormat := f.TimestampFormat
```
Comment (aybabtme) [inline text_formatter.go]: this isn't a race

Comment (sp-marcos-quesada) [inline text_formatter.go]: Passing our test suite using --race it always fails at the same point, we need to get ride of all race conditions, so,  that's why we decide to fork the package:

WARNING: DATA RACE
Read by goroutine 12:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()   [[[READS .TimestampFormat ]]]
      /home/travis/gopath/src/github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/text_formatter.go:76 +0x42c
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/travis/gopath/src/github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/entry.go:41 +0xaa
/httpport.go:145 +0x600
Previous write by goroutine 9:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()   [[[ASSIGNS f.TimestampFormat = DefaultTimestampFormat]]]
      /home/travis/gopath/src/github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/text_formatter.go:77 +0x44c
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/travis/gopath/src/github.com/socialpoint/monstercity-sprocket/Godeps/_workspace/src/github.com/Sirupsen/logrus/entry.go:41 +0xaa 
....

Comment (aybabtme) [inline text_formatter.go]: Can you share what code generates that race report, or a reproducer?

Comment (sp-marcos-quesada) [inline text_formatter.go]: I cannot share with you that code, it's a private repo, to reproduce it i've made a simple example:
> cat logRc.go 

package main

import(
	"github.com/Sirupsen/logrus" 	
)

func main() {
	var log = logrus.New()

	for i:=0; i<10; i++ {
		go func(){
			log.Info("foo")
		}()
	}
}


> go run --race logRc.go

INFO[0000] foo                                          
==================
WARNING: DATA RACE
Read by goroutine 7:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()
      /home/marcos/go/src/github.com/Sirupsen/logrus/text_formatter.go:76 +0x787
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:41 +0xa1
  github.com/Sirupsen/logrus.(*Entry).log()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:84 +0x351
  github.com/Sirupsen/logrus.(*Entry).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:119 +0xad
  github.com/Sirupsen/logrus.(*Logger).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logger.go:122 +0x15d
  main.main.func1()
      /home/marcos/go/logRc.go:12 +0xfb

Previous write by goroutine 6:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()
      /home/marcos/go/src/github.com/Sirupsen/logrus/text_formatter.go:77 +0x7a7
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:41 +0xa1
  github.com/Sirupsen/logrus.(*Entry).log()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:84 +0x351
  github.com/Sirupsen/logrus.(*Entry).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:119 +0xad
  github.com/Sirupsen/logrus.(*Logger).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logger.go:122 +0x15d
  main.main.func1()
      /home/marcos/go/logRc.go:12 +0xfb

Goroutine 7 (running) created at:
  main.main()
      /home/marcos/go/logRc.go:13 +0x217

Goroutine 6 (finished) created at:
  main.main()
      /home/marcos/go/logRc.go:13 +0x217
==================
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
INFO[0000] foo                                          
Found 1 data race(s)
exit status 66

**Comment by sp-ronny-lopez** [comment]
@sp-marcos-quesada I was not able to reproduce it. Here is a PR that runs the test suite with `-race` and builds all passing: https://github.com/Sirupsen/logrus/pull/315

**Comment by sp-marcos-quesada** [comment]
I've already checked your test, and I still detecting the same Race, here's de log:

 ✘  ➦ 21d4508 ± GOMAXPROCS=4 got -race -run TestLoggingRace 
=== RUN   TestLoggingRace
==================
WARNING: DATA RACE
Read by goroutine 85:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()
      /home/marcos/go/src/github.com/Sirupsen/logrus/text_formatter.go:76 +0x787
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:41 +0xa1
  github.com/Sirupsen/logrus.(*Entry).log()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:84 +0x351
  github.com/Sirupsen/logrus.(*Entry).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:119 +0xad
  github.com/Sirupsen/logrus.(*Logger).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logger.go:122 +0x15d
  github.com/Sirupsen/logrus.TestLoggingRace.func1()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logrus_test.go:311 +0xfb

Previous write by goroutine 74:
  github.com/Sirupsen/logrus.(*TextFormatter).Format()
      /home/marcos/go/src/github.com/Sirupsen/logrus/text_formatter.go:77 +0x7a7
  github.com/Sirupsen/logrus.(*Entry).Reader()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:41 +0xa1
  github.com/Sirupsen/logrus.(*Entry).log()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:84 +0x351
  github.com/Sirupsen/logrus.(*Entry).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/entry.go:119 +0xad
  github.com/Sirupsen/logrus.(*Logger).Info()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logger.go:122 +0x15d
  github.com/Sirupsen/logrus.TestLoggingRace.func1()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logrus_test.go:311 +0xfb

Goroutine 85 (running) created at:
  github.com/Sirupsen/logrus.TestLoggingRace()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logrus_test.go:313 +0x26e
  testing.tRunner()
      /tmp/workdir/go/src/testing/testing.go:456 +0xdc

Goroutine 74 (running) created at:
  github.com/Sirupsen/logrus.TestLoggingRace()
      /home/marcos/go/src/github.com/Sirupsen/logrus/logrus_test.go:313 +0x26e
  testing.tRunner()
      /tmp/workdir/go/src/testing/testing.go:456 +0xdc
==================
time="2016-01-23T11:53:40+01:00" level=info msg=info 
time="2016-01-23T11:53:40+01:00" level=info msg=info 
time="2016-01-23T11:53:40+01:00" level=info msg=info 
time="2016-01-23T11:53:40+01:00" level=info msg=info 
....

**Comment by sp-marcos-quesada** [comment]
Sorry @aybabtme , it was my fault! Upgrading logrus version made race conditions disappear. Thank your for your time, and sorry again!

**Pull request closed without merging**